junifer/api/queue_context/queue_context_adapter.py

```diff
@@ -19,14 +19,6 @@ class QueueContextAdapter(ABC):
 
     """
 
-    @abstractmethod
```

Comment:
There's no pre-run anymore?

What if the user wants to set environmental variables even in a gnu parallel environment? Keep in mind that GNU parallel can also be used to run over SSH.

I liked the idea of having some sort of self-contained running scripts. 

Imagine this scenario:

- User queues with `local` and starts running gnu-parallel on the HCP. 
- Takes too long, wants to compute another thing, presses CTRL+C to stop.
- Runs another thing, closes the terminal.
- Opens the terminal, goes to the `junifer_jobs` dir and runs the `parallel` command again.

This should work.

Reply:
For the SSH, not officialy, but someone might "tweak" the parallel command. In any case, even if SSH is not a supported use case, just closing the terminal and opening it again should continue were it was.

Reply:
> For the SSH, not officialy, but someone might "tweak" the parallel command.

That's for the "hacker" spirited-soul :)

>  In any case, even if SSH is not a supported use case, just closing the terminal and opening it again should continue were it was.

Of course.

Reply:
My own personal cluster (my phd lab during the nights).

![de noche](https://github.com/juaml/junifer/assets/4493699/4ce62d20-d43b-437f-823f-0b35a1c6062e)